eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/admin/controller/HttpHandlerManager.java

```diff
@@ -18,48 +18,231 @@
+ * This class manages the registration of {@linkplain com.sun.net.httpserver.HttpHandler HttpHandler} for an {@linkplain
+ * com.sun.net.httpserver.HttpServer HttpServer}.
  */
 
+@Slf4j
 public class HttpHandlerManager {
 
     private final List<HttpHandler> httpHandlers = new ArrayList<>();
 
+    private final Map<String, HttpHandler> httpHandlerMap = new ConcurrentHashMap<>();
+
     /**
      * Registers an HTTP handler.
      *
-     * @param httpHandler The {@link HttpHandler} to be registered.
-     *                    A handler which is invoked to process HTTP exchanges.
-     *                    Each HTTP exchange is handled by one of these handlers.
+     * @param httpHandler The {@link HttpHandler} to be registered. A handler which is invoked to process HTTP exchanges. Each HTTP exchange is
+     *                    handled by one of these handlers.
      */
     public void register(HttpHandler httpHandler) {
         this.httpHandlers.add(httpHandler);
     }
 
-    /**
-     * Registers multiple HTTP handlers to a given HttpServer.
-     * <p>
-     * Each HTTP handler is annotated with the {@link EventHttpHandler} annotation,
-     * which specifies the path where the handler should be registered.
-     *
-     * @param server A HttpServer object that is bound to an IP address and port number
-     *               and listens for incoming TCP connections from clients on this address.
-     *               The registered HTTP handlers will be associated with this server.
-     */
-    public void registerHttpHandler(HttpServer server) {
-        httpHandlers.forEach(httpHandler -> {
-            EventHttpHandler eventHttpHandler = httpHandler.getClass().getAnnotation(EventHttpHandler.class);
-            server.createContext(eventHttpHandler.path(), httpHandler);
+    public void register() {
+        httpHandlers.forEach(e -> {
+            EventHttpHandler eventHttpHandler = e.getClass().getAnnotation(EventHttpHandler.class);
+            httpHandlerMap.putIfAbsent(eventHttpHandler.path(), e);
         });
+    }
 
+    public void exec(ChannelHandlerContext ctx, HttpRequest httpRequest) {
+        String uriStr = httpRequest.uri();
+        URI uri = URI.create(uriStr);
+        HttpHandler httpHandler = httpHandlerMap.get(uri.getPath());
+        if (httpHandler != null) {
+            try {
+                HttpHandlerManager.AdminHttpExchange adminHttpExchange = new HttpHandlerManager.AdminHttpExchange(ctx, httpRequest);
+                httpHandler.handle(adminHttpExchange);
+                adminHttpExchange.writeAndFlash();
+                return;
+            } catch (Exception e) {
+                log.error(e.getMessage(), e);
+                ctx.writeAndFlush(HttpResponseUtils.createInternalServerError()).addListener(ChannelFutureListener.CLOSE);
+            }
+        } else {
+            ctx.writeAndFlush(HttpResponseUtils.createNotFound()).addListener(ChannelFutureListener.CLOSE);
+        }
```

Comment:
How about integrate `EventMeshAdminServer` with `HandlerService` instead of implementing a group of http processing logic seperately for admin?